include/sys/fs/zfs.h

```diff
@@ -741,6 +744,24 @@ typedef enum zpool_errata {
 	ZPOOL_ERRATA_ZOL_2094_ASYNC_DESTROY,
 } zpool_errata_t;
 
+#ifndef BIT /* Include BIT() for userspace */
+#define	BIT(n)	(1ULL << (n))
+#endif
+
+enum vdev_stat_flags {VS_DEFAULT = 0, VS_LATENCY = 1, VS_QUEUES = 2,
+		VS_HISTO = 3, VS_FLAGS_COUNT /* always last element */};
+
+/* Mask versions of all our flags */
+#define	VS_DEFAULT_M	BIT(VS_DEFAULT)
+#define	VS_LATENCY_M	BIT(VS_LATENCY)
+#define	VS_QUEUES_M	BIT(VS_QUEUES)
+#define	VS_HISTO_M	BIT(VS_HISTO)
```

Comment:
Unfortunately, due to coalescing (see vdev_queue_aggregate()), vdev-level performance is less useful than disk or vfs level. It becomes very difficult to correlate a vfs-level request to one or more disk I/Os, and visibility at the vdev-level doesn't help much. That said, I do like the approach and think that we can reuse this work at other levels. Thanks